components/BillTestimonies/BillTestimonies.js

```diff
@@ -1,21 +1,28 @@
 import React from "react";
-import { testimonies } from "../MockTestimonies"
 import { Table, Container } from 'react-bootstrap'
 import ExpandTestimony from "../ExpandTestimony/ExpandTestimony";
+import { usePublishedTestimonyListing } from "../db";
+
 
 // the word "testimonies": In more general, commonly used, contexts, the plural form will also be testimony.  However, in more specific contexts, the plural form can also be testimonies e.g. in reference to various types of testimonies or a collection of testimonies.
 
+
 const ViewTestimonies = (props) => {
   const bill = props.bill
-  const testimoniesComponent = !testimonies ? "" :
+  const testimoniesResponse = usePublishedTestimonyListing({billId: bill.id})
+
+  const testimonies = testimoniesResponse.status == "loading" ? [] : testimoniesResponse.result
```

Comment:
If there is an error, status will be "error" and result will be undefined, which I think will cause an error below.

I think we can change this to `const testimonies = testimoniesResponse.status == "success" ? testimoniesResponse.result : []` so testimonies is an empty array if loading or errored.

Reply:
yes, I had not thought about the error state on this one.  I will update.